scylla-cql/src/frame/response/result.rs

```diff
@@ -1027,6 +1087,13 @@ fn deser_result_metadata(
 
     let paging_state = PagingStateResponse::new_from_raw_bytes(raw_paging_state);
 
+    let new_metadata_id = metadata_changed
+        .then(|| {
+            types::read_short_bytes(buf).map_err(ResultMetadataParseError::NewMetadataIdParseError)
+        })
+        .transpose()?
+        .map(|x| cow_bytes::CowBytes::from(x).into_owned());
```

Comment:
:thought_balloon: So here we allocate if only new metadata id is present. Am I correct that it should happen rarely?

Even if that's the case, I'd love to introduce a new flavor of at least `read_short_bytes` which would operate on a `FrameSlice` instead of plain byte slices. Then the allocation would never be needed here.